tests/shardddl3/run.sh

```diff
@@ -873,6 +873,1442 @@ function DM_112() {
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb1} add column (b int, c int);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb2} add column (b int, c int);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(10,10,10);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(11,11,11);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(12,12,12);"
+
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+# Add multiple fileds to a single table.
+function DM_113 {
+    run_case 113 "double-source-pessimistic" "init_table 111 211 212" "clean_table" "pessimistic"
+    run_case 113 "double-source-optimistic" "init_table 111 211 212" "clean_table" "optimistic"
+}
+
+function DM_114_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3,3);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} drop column b, drop column c;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6,6);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb1} drop column b, drop column c;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9,9);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb2} drop column b, drop column c;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(10);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(11);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(12);"
+
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+# Drop multiple fields from a single table.
+function DM_114 {
+    run_case 114 "double-source-pessimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, b int, c int);\"" \
+    "clean_table" "pessimistic"
+    run_case 114 "double-source-optimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, b int, c int);\"" \
+    "clean_table" "optimistic"
+}
+
+function DM_115_CASE {
```

Comment:
@lichunzhu this test is marked as negative in our spreadsheet, I think it's positive in fact